pkg/provisioner/ironic/ironic.go

```diff
@@ -363,23 +363,28 @@ func (p *ironicProvisioner) ValidateManagementAccess(data provisioner.Management
 			return
 		}
 
-		ironicNode, err = nodes.Create(
-			p.client,
-			nodes.CreateOpts{
-				Driver:              bmcAccess.Driver(),
-				BootInterface:       bmcAccess.BootInterface(),
-				Name:                p.objectMeta.Name,
-				DriverInfo:          driverInfo,
-				DeployInterface:     p.deployInterface(data.CurrentImage),
-				InspectInterface:    "inspector",
-				ManagementInterface: bmcAccess.ManagementInterface(),
-				PowerInterface:      bmcAccess.PowerInterface(),
-				RAIDInterface:       bmcAccess.RAIDInterface(),
-				VendorInterface:     bmcAccess.VendorInterface(),
-				Properties: map[string]interface{}{
-					"capabilities": bootModeCapabilities[data.BootMode],
-				},
-			}).Extract()
+		opts := nodes.CreateOpts{
+			Driver:              bmcAccess.Driver(),
+			BootInterface:       bmcAccess.BootInterface(),
+			Name:                p.objectMeta.Name,
+			DriverInfo:          driverInfo,
+			DeployInterface:     p.deployInterface(data.CurrentImage),
+			InspectInterface:    "inspector",
+			ManagementInterface: bmcAccess.ManagementInterface(),
+			PowerInterface:      bmcAccess.PowerInterface(),
+			RAIDInterface:       bmcAccess.RAIDInterface(),
+			VendorInterface:     bmcAccess.VendorInterface(),
+			Properties: map[string]interface{}{
+				"capabilities": bootModeCapabilities[data.BootMode],
+			},
+		}
+
+		if bmcAccess.BIOSInterface() != "" {
+			// Only override Ironic defaults if explicitly set
+			opts.BIOSInterface = bmcAccess.BIOSInterface()
+		}
```

Comment:
Q: Is this really required this check (ie, if `bmAccess.BIOSInterface()` is not set then `opts.BIOSInterface` will be an empty string, is there any issue for that in gophercloud/ironic api maybe)?

Reply:
Yes this is intentional and the way it currently works. If opts.BIOSInterface is set to "" then BIOSInterface will be omitted by Gophercloud - https://github.com/gophercloud/gophercloud/blob/master/openstack/baremetal/v1/nodes/requests.go#L189 and the Ironic default will be used. We only need to override the default when the default that Ironic picks is not desired.

Reply:
According to that link, `opts.BIOSInterface` is a `string` (not `*string`), so if it is not set then it will get the zero value for a string (which is `""`). So this check has no effect.

Reply:
Thanks. Fixed.